helper/common/common.go

```diff
@@ -30,6 +32,16 @@ var (
 	errInvalidDuration = errors.New("invalid duration")
 )
 
+// RetryForever will execute a function until it completes without error
+func RetryForever(ctx context.Context, interval time.Duration, fn func(context.Context) error) {
```

Comment:
Do we need to have context as a parameter here?

Reply:
Yes a valid use case is to pass in a `fn` which will exit with err == nil on `ctx.Done()` of the ctx passed in here

Reply:
Ok, just make it sure that we are aligned with these chatgpt suggestions

- `Error handling`: In your code, you're converting the error returned by the function fn into a retry.RetryableError if it fails. This implies that any error encountered during the execution of fn should be considered retryable. Make sure this behavior aligns with your requirements and that you're not inadvertently retrying non-retryable errors.
- `Context usage`: The provided ctx parameter is used for cancellation. Ensure that you propagate the cancellation appropriately within the fn function to avoid unnecessary retries when the context is canceled.

Reply:
Thanks for explanation. I did not know that wrapping error with `retry.RetryableError(err)` allows `go-retry` library to differentiate between errors that should be retried  and errors that should not be retried. you can resolved